app/controllers/mccloud/IsPublicServicePensionsRemedyController.scala

```diff
@@ -0,0 +1,138 @@
+                                                        userAnswersService: UserAnswersService,
+                                                        navigator: CompoundNavigator,
+                                                        identify: IdentifierAction,
+                                                        getData: DataRetrievalAction,
+                                                        allowAccess: AllowAccessActionProvider,
+                                                        requireData: DataRequiredAction,
+                                                        formProvider: YesNoFormProvider,
+                                                        val controllerComponents: MessagesControllerComponents,
+                                                        chargeServiceHelper: ChargeServiceHelper,
+                                                        config: FrontendAppConfig,
+                                                        renderer: Renderer)(implicit ec: ExecutionContext)
+  extends FrontendBaseController
+    with I18nSupport
+    with NunjucksSupport {
+
+  private def form(memberName: String)(implicit messages: Messages): Form[Boolean] =
+    formProvider(messages("isPublicServicePensionsRemedy.error.required", memberName))
+
+  def onPageLoad(chargeType: ChargeType,
+                 mode: Mode,
+                 srn: String,
+                 startDate: LocalDate,
+                 accessType: AccessType,
+                 version: Int,
+                 index: Index): Action[AnyContent] =
+    (identify andThen getData(srn, startDate) andThen requireData andThen allowAccess(srn, startDate, None, version, accessType)).async { implicit request =>
+      DataRetrievals.retrieveSchemeName { schemeName =>
+
+        val chargeTypeDescription = Messages(s"isPublicServicePensionsRemedy.chargeType.${chargeType.toString}")
+
+        val viewModel = GenericViewModel(
+          submitUrl = routes.IsPublicServicePensionsRemedyController.onSubmit(chargeType, mode, srn, startDate, accessType, version, index).url,
+          returnUrl = controllers.routes.ReturnToSchemeDetailsController.returnToSchemeDetails(srn, startDate, accessType, version).url,
+          schemeName = schemeName
+        )
+
+        val preparedForm = request.userAnswers.get(IsPublicServicePensionsRemedyPage(chargeType, index)) match {
+          case None => form(chargeTypeDescription)
+          case Some(value) => form(chargeTypeDescription).fill(value)
+        }
+
+        val json = Json.obj(
+          "srn" -> srn,
+          "startDate" -> Some(localDateToString(startDate)),
+          "form" -> preparedForm,
+          "viewModel" -> viewModel,
+          "radios" -> Radios.yesNo(preparedForm("value")),
+          "chargeTypeDescription" -> Messages(s"isPublicServicePensionsRemedy.chargeType.${chargeType.toString}")
```

Comment:
Could you use the val you created on 72 here as well?

Reply:
Well spotted. I've fixed this.